### DISCUSSION TITLE
Stripped build now provided

### DISCUSSION THREAD

**Discussion by rmast** [rmast/revealshapes#1] [Announcements]
@jsbien I today composed a minimal git repo that builds a revealshapes-executable for ubuntu 2004 and 2204.
This version gives a message when the djvu-file is not provided and lists the coordinates to stdout. Only the -a option was not well taken over by chatgpt or tested. You can download such an executable from the Actions menue.

**Comment by jsbien** [comment]
Great! 

Tomorrow I will try to convert the output to a djview-poliqarp index. I'm sorry to say I don't know what blit is in this context -(
Wikipedia (https://en.wikipedia.org/wiki/Bit_blit) is not helpful in this respect.

FYI, when I tried to compile myself (autogen, configure, make) I got

`revealshapes.cpp:32:10: fatal error: DjVuDocument.h: No such file or directory`
`   32 | #include "DjVuDocument.h"`

**Comment by jsbien** [comment]
Looks like the conversion rule can be
`<file-name> d-<blit number>-<blit shapeno>;file:<file-name>?djvuopts=&highlight=<left>,<blit bottom>,<shape bit columns>,<rows>&page=<page number + 1>; <file-name> d-<blit number>-<blit shapeno>;`
The second element is a must, the other elements are more or less arbitrary, but I hope this form would be useful.
What you be so kind to provide also output in this form, preferably in a separate file?
I can try to write a Python program to convert your present output to this form, but I'm very slow programmer :-(
As you can see, I have no use for `row size`. What is it for? Am I missing something?

**Comment by rmast** [comment]
"Augezdecki-01 001:00000 [2l]";file:../../DjVu/Augezdecki-01.djvu?djvuopts=&highlight=00047,00925,00064,00073&page=001;Augezdecki-01.djvu p 1 x 47 y 925 h 64 w 73;※ 0_Augezdecki-01_PT08_402.djvu 00002 (2) new
"Augezdecki-01 001:00001 [2»]";file:../../DjVu/Augezdecki-01.djvu?djvuopts=&highlight=00136,00932,00071,00069&page=001;Augezdecki-01.djvu p 1 x 136 y 932 h 71 w 69;※ 0_Augezdecki-01_PT08_402.djvu 00003 (3) new

I see differences in the filenames, the first doesn't have either a path or an extension like the second. The third has 0_ as prefix and then _PT08_402 before the extension. So these might be different filenames. Revealing the right general dictionary-name from a djvu with multiple 20 page dictionaries might be a challenge on it's own. The examples using djvulibre usually don't take such a situation into account. Exportshapes assumes there's only one such dictionary in the djvu.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jsbien** [comment]
> the first doesn't have either a path or an extension like the second
The first is just an identifier of a font, only the second is the file.

> The third has 0_ as prefix and then _PT08_402 before the extension. 
This is not a file but the directory name created by djvudict which I preserve in the index just in case.

By `<file name>` I mean just the argument of the revealshapes call.

So for the call, say, `./revealshapes Augezdecki-01a_PT08_403.djvu` (FYI, PT stands for _Polonia Typographica_ , 8 is the fascicule number and 403 is the plate number) I would be happy with the file `Augezdecki-01a_PT08_403.csv` containing lines like

`"Augezdecki-01a_PT08_403 d-0-0";file:Augezdecki-01a_PT08_403.djvu?djvuopts=&highlight=00369,00204,00063,00073&page=001;Augezdecki-01a_PT08_403.djvu d-0-0;`

**Comment by rmast** [comment]
The [2l] behind the first filename, describing the character that was found on that position in the searchable text, won't you miss that?

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
No, it's the result of OCR by Tesseract, which is not reliable for such old texts, but perhaps can be useful to group similar characters together. I can add it myself.

**Comment by rmast** [comment]
A djvu-document page contains blits and shapes, with blits pointing to a shapeno.

The current output of revealshapes is one line per blit.
Blits describe the positions where shapes were found, and the shape number points to the shared and local dictionaries, depending on the shape count of the shared dictionary.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
Thanks.

**Comment by rmast** [comment]
At the end of the line the numbers are not fully equal to blitnr.

https://github.com/jsbien/early_fonts_inventory/blob/main/font_tables/indexes/Augezdecki-01/Augezdecki-01.csv

has 2 equal lines on the bottom with the right number going up. The right number on top is already higher than the line number it is on. What is that "new" for?

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jsbien** [comment]
"new" is copied from the djvudict output such as
`Records jb2_new_symbol_add_to_image_and_library:	id: 0	x: 47	y: 998	`
 I don't know for sure what it means.

The strange lines you noticed may is a bug in the program used to convert djvudict action.log to index, the last line is simply redundant. I will not correct it as I think that thanks to revealshapes I no longer need djvudict.

**Comment by rmast** [comment]
Most of that index is comment/Description?

![image](https://github.com/rmast/revealshapes/assets/3341558/18382e51-6dac-4dff-9080-86ecc7943eb8)

**Reply by jsbien** [reply]
Yes. BTW, I will have no access to Internet for several days, so please don't be surprised by a delay in answers.
